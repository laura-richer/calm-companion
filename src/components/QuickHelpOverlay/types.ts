export type QuickHelpOverlayProps = {
  show: boolean;
  itemId?: number;
  onClose: () => void;
};
