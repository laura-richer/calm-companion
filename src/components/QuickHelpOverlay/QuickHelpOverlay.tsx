import React from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '~store/index';
import { selectItemById } from '~store/slices/quickHelp';
import FullHeightOverlayWrapper from '~components/FullHeightOverlayWrapper/FullHeightOverlayWrapper';
import RteWrapper from '~components/Rte/RteWrapper';
import { QuickHelpOverlayProps } from './types';

const QuickHelpOverlay = ({ show, onClose, itemId }: QuickHelpOverlayProps) => {
  const currentItem = useSelector((state: RootState) => selectItemById(state, itemId));

  return (
    <React.Fragment>
      {currentItem ? (
        <FullHeightOverlayWrapper show={show} onClose={onClose} title={currentItem.title}>
          <RteWrapper content={currentItem.content} />
        </FullHeightOverlayWrapper>
      ) : null}
    </React.Fragment>
  );
};

export default QuickHelpOverlay;
