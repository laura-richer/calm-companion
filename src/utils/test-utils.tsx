import React, { PropsWithChildren } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render, RenderOptions } from '@testing-library/react-native';
import { Provider as StoreProvider } from 'react-redux';
import { EnhancedStore, configureStore } from '@reduxjs/toolkit';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { RootState } from '~store/index';
import navigationReducer, {
  initialState as navigationInitialState,
} from '~store/slices/navigation';
import quickHelpReducer, { initialState as quickHelpInitialState } from '~store/slices/quickHelp';

type DeepPartial<T> = {
  [K in keyof T]?: T[K] extends object ? DeepPartial<T[K]> : T[K];
};

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: EnhancedStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  { preloadedState = {}, store = createTestStore(preloadedState) }: ExtendedRenderOptions = {}
) => {
  const wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => {
    return (
      <StoreProvider store={store}>
        <SafeAreaProvider>
          <NavigationContainer>{children}</NavigationContainer>
        </SafeAreaProvider>
      </StoreProvider>
    );
  };

  return {
    ...render(ui, {
      wrapper,
    }),
  };
};

const _createStore = (preloadedState: Partial<RootState> = {}): EnhancedStore => {
  return configureStore({
    reducer: {
      navigation: navigationReducer,
      quickHelp: quickHelpReducer,
    },
    preloadedState,
    middleware: getDefaultMiddleware =>
      getDefaultMiddleware({
        immutableCheck: { warnAfter: 256 },
        serializableCheck: { warnAfter: 256 },
      }),
  });
};

export const createTestStore = (preloadedState: DeepPartial<RootState> = {}): EnhancedStore => {
  const state: RootState = {
    // @ts-ignore
    navigation: {
      ...navigationInitialState,
      ...(preloadedState.navigation || {}),
    },
    // @ts-ignore
    quickHelp: {
      ...quickHelpInitialState,
      ...(preloadedState.quickHelp || {}),
    },
  };
  const store: EnhancedStore = _createStore(state);
  const origDispatch = store.dispatch;
  store.dispatch = jest.fn(origDispatch);
  return store;
};
