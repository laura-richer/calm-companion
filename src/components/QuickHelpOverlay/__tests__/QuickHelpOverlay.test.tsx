import React from 'react';
import { screen } from '@testing-library/react-native';
import { createTestStore, renderWithProviders } from '~utils/test-utils';
import QuickHelpOverlay from '../QuickHelpOverlay';

const store = createTestStore();

const _renderDefault = () => {
  return renderWithProviders(<QuickHelpOverlay show={true} onClose={() => {}} itemId={1} />, {
    store,
  });
};
describe('quick help overlay overlay', () => {
  beforeEach(() => {
    _renderDefault();
  });

  test('matches snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  // test('renders overlay if current item exists', () => {
  //   const content = screen.getByText('Child content');
  //   expect(content).toBeOnTheScreen();
  // });

  // test('doesnt render overlay if current item is undefined', () => {
  //   const title = screen.getByText('Full height modal');
  //   expect(title).toBeOnTheScreen();
  // });
});
