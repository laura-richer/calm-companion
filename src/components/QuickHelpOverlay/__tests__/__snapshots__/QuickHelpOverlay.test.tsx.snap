// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quick help overlay overlay matches snapshot 1`] = `null`;
